app/models/ZkKafka.scala

```diff
@@ -52,7 +55,7 @@ object ZkKafka {
     val maybeData = Option(zkClient.getData.forPath(path))
 
     // log a message if we get a null result for a path read
-    maybeData.map(Option(_)).filter(_.isDefined).foreach(_ => Logger.warn(s"null data retrieved for path $path"))
+    maybeData.map(Option(_)).filter(_.isEmpty).foreach(_ => Logger.warn(s"null data retrieved for path $path"))
```

Comment:
You are right, this is just a strange pattern.  There is probably a more clear way to express it, but I don't know it off the top of my head.